perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -254,6 +256,31 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
 		return attributes;
 	}
 
+	public List<Attribute> getAttributes(PerunSession sess, Group group, Resource resource, Member member, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException {
+		List<Attribute> attributes = getAttributes(sess, resource, member, attrNames, workWithUserAttributes);
+
+		if(attrNames.isEmpty()) {
+			attributes.addAll(getAttributes(sess, member, group));
+			return attributes;
+		}
+
+		List<String> memberGroupAttributeNames = new ArrayList<>();
+
+		//possible inefficiency - additional iteration over attrNames
+		for(String attributeName: attrNames) {
+			if(attributeName.startsWith(AttributesManager.NS_MEMBER_GROUP_ATTR)) {
```

Comment:
You can simplify this by using only positive branch. Because rest of attribute namespaces are already retrieved by above code (if possible and present in params), it makes no sense to log error "unsupported" in else branch when method as a whole returns it. Lists of attribute names are expected to be reasonable short, so iterating over it is OK.